src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -33,28 +43,152 @@ const ReimbursementRequestTable = ({
+      return 1;
+    }
+    return 0;
+  };
+
+  const statusDescendingComparator = (a: ReimbursementStatusType, b: ReimbursementStatusType) => {
+    const statusOrder = new Map<ReimbursementStatusType, number>([
+      [ReimbursementStatusType.PENDING_FINANCE, 1],
+      [ReimbursementStatusType.SABO_SUBMITTED, 2],
+      [ReimbursementStatusType.ADVISOR_APPROVED, 3],
+      [ReimbursementStatusType.REIMBURSED, 4],
+      [ReimbursementStatusType.DENIED, 5]
+    ]);
+
+    const bConverted = statusOrder.get(b);
+    const aConverted = statusOrder.get(a);
+
+    if (bConverted !== undefined && aConverted !== undefined) {
+      if (bConverted < aConverted) {
+        return -1;
+      }
+      if (bConverted > aConverted) {
+        return 1;
+      }
+    }
+    return 0;
+  };
+
+  const vendorDescendingComparator = (a: Vendor, b: Vendor) => {
+    if (b.name < a.name) {
+      return -1;
+    }
+    if (b.name > a.name) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const submitterDescendingComparator = (a: User, b: User) => {
+    if (b.firstName < a.firstName) {
+      return -1;
+    }
+    if (b.firstName > a.firstName) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const rows = displayedReimbursementRequests.map(createReimbursementRequestRowData).sort((a, b) => {
+    if (orderBy === 'vendor') {
+      return !isAscendingOrder
+        ? vendorDescendingComparator(a.vendor, b.vendor)
+        : -vendorDescendingComparator(a.vendor, b.vendor);
+    }
+    if (orderBy === 'status') {
+      return !isAscendingOrder
+        ? statusDescendingComparator(a.status, b.status)
+        : -statusDescendingComparator(a.status, b.status);
+    }
+    if (orderBy === 'submitter') {
+      return !isAscendingOrder
+        ? submitterDescendingComparator(a.submitter, b.submitter)
+        : -submitterDescendingComparator(a.submitter, b.submitter);
+    }
+    if (b[orderBy] === undefined) {
+      return -1;
+    }
+    return !isAscendingOrder ? descendingComparator(a, b, orderBy) : -descendingComparator(a, b, orderBy);
+  });
 
   const tabs = [{ label: 'My Requests', value: 0 }];
   if (canViewAllReimbursementRequests) tabs.push({ label: 'All Club Requests', value: 1 });
 
+  const handleRequestSort = (property: keyof ReimbursementRequestRow) => {
+    setAscendingOrder(!isAscendingOrder);
+    setOrderBy(property);
+  };
+
+  const headCells: readonly ReimbursementTableHeadCell[] = [
+    {
+      id: 'submitter',
+      label: 'Recipient'
+    },
+    {
+      id: 'saboId',
+      label: 'Sabo ID'
+    },
+    {
+      id: 'amount',
+      label: 'Amount ($)'
+    },
+    {
+      id: 'dateSubmitted',
+      label: 'Date Submitted'
+    },
+    {
+      id: 'dateSubmittedToSabo',
+      label: 'Date Submitted To Sabo'
+    },
+    {
+      id: 'vendor',
+      label: 'Vendor'
+    },
+    {
+      id: 'refundSource',
+      label: 'Refund Source'
+    },
+    {
+      id: 'status',
+      label: 'Status'
+    }
+  ];
```

Comment:
per my comment below, you should be able to incorporate these ids into our ColumnHeader component, so you shouldn't' need this list

Reply:
The previous implementation of the column headers manually imputed the labels and the names. I wanted to use a list so I could just use .map to add the labels (this was also done in the example in the MaterialUI Docs).